### PR TITLE
CFE-2507: Policy function storejson() no longer truncates strings lager than 4096 bytes

### PR DESCRIPTION
**Pull request by larsewi**
Policy function storejson() no longer truncates strings lager than 4096 bytes

### CODE REVIEW

File: libpromises/evalfunction.c [hunk 1]
```diff
@@ -7325,21 +7325,15 @@ static FnCallResult FnCallStoreJson(EvalContext *ctx, ARG_UNUSED const Policy *p
...
-
-    char buf[CF_BUFSIZE];
-    snprintf(buf, CF_BUFSIZE, "%s", StringWriterData(w));
+    const uint32_t buf_size = strlen(StringWriterData(w));
+    char buf[buf_size + 1];
+    strcpy(buf, StringWriterData(w));
+    
     WriterClose(w);
 
     return FnReturn(buf);
```
Comment (olehermanse) [inline libpromises/evalfunction.c:7339]: I don't think you even need `buf` at all, I think you can do:

```
return FnReturnNoCopy(StringWriterClose(w));
```

File: tests/acceptance/01_vars/02_functions/storejson_edge_case.cf [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+bundle agent check {
+  methods:
+      "check"  usebundle => dcs_check_state(test,
+                                            "$(this.promise_filename).expected.json",
+                                            $(this.promise_filename));
+}
```
Comment (olehermanse) [inline tests/acceptance/01_vars/02_functions/storejson_edge_case.cf:22]: Missing newline at end of file.

**Review by olehermanse — APPROVED** [review]
Great!

**Comment by larsewi** [comment]
@cf-bottom jenkins, please!

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/5971/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/5971/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-5971/

**Review by vpodzime — APPROVED** [review]
Looks good to me.

**Comment by olehermanse** [comment]
Cherry-picks: 
https://github.com/cfengine/core/pull/4422
https://github.com/cfengine/core/pull/4423

**Pull request merged**